### DISCUSSION TITLE
Is anyone working on a Rust impl of the query language, and if not, would you be interested in someone doing that?

### DISCUSSION THREAD

**Discussion by alilleybrinker** [kdl-org/kdl#397] [General]
Basically the title. It looks like the `kdl` crate doesn't have an impl of the KDL query language, which would be quite useful for a project of mine. I'm happy to take a crack at it as a PR on the `kdl` crate, but if someone else is already doing it I don't want to duplicate the effort.

**Comment by zkat** [comment]
The prerelease of `kdl@5.0` includes a query API: https://docs.rs/kdl/5.0.0-alpha.1/kdl/struct.KdlDocument.html#method.query

The next release of the kdl crate, which is going to be the version that supports kdlv2, also will have a more mature and sussed-out query API.

**Reply by alilleybrinker** [reply]
Oh excellent! Glad I asked before I attempted to do it myself :smile:
